tests/e2e/utils/src/page-utils.js

```diff
@@ -209,6 +209,38 @@ const selectOptionInSelect2 = async ( value, selector = 'input.select2-search__f
 	await page.keyboard.press('Enter');
 };
 
+/**
+ * Apply a coupon code within cart or checkout.
+ * Method will try to apply a coupon in the checkout, otherwise will try to apply in the cart.
+ *
+ * @param couponCode string
+ * @returns {Promise<void>}
+ */
+const applyCoupon = async ( couponCode ) => {
+	try {
+		await expect(page).toClick('a', {text: 'Click here to enter your code'});
+		await uiUnblocked();
+		await clearAndFillInput('#coupon_code', couponCode);
+		await expect(page).toClick('button', {text: 'Apply coupon'});
+		await uiUnblocked();
+	} catch (error) {
+		await clearAndFillInput('#coupon_code', couponCode);
+		await expect(page).toClick('button', {text: 'Apply coupon'});
+		await uiUnblocked();
+	};
+};
+
+/**
+ * Remove one coupon within cart or checkout.
+ *
+ * @returns {Promise<void>}
+ */
+const removeCoupon = async () => {
+	await expect(page).toClick('.woocommerce-remove-coupon', {text: '[Remove]'});
```

Comment:
Could we use something like the `data-coupon` identifier here to specify the coupon `couponCode` we are removing in the event of there being multiple coupons in use.

Reply:
Hey, you were totally right! I just updated `removeCoupon` method so it includes the coupon name.